src/Data/Filterable.purs

```diff
@@ -127,3 +128,24 @@ instance filterableEither :: Monoid m => Filterable (Either m) where
 
   filter p = filterDefault p
 
+instance filterableList :: Filterable List.List where
+  -- partitionMap :: forall a l r. (a -> Either l r) -> List a -> { left :: List l, right :: List r }
+  partitionMap p xs = foldl select { left: List.Nil, right: List.Nil } xs
```

Comment:
okay with me - i first wrote it as foldr and changed it to foldl actually (hence the bug you spotted). will do that now.